### DISCUSSION TITLE
Are we allowed to reset the label color for the header field in story card Pass?

### DISCUSSION THREAD

**Discussion by SamYueHW** [alexandercerutti/passkit-generator#191] [Q&A] [answered]
Hi,
Is it possible that we can reset/overwrite the label color for the header field in story card Pass. 
Because when I trying to push the passData:

```javascript
newPass.headerFields.push({
    "key": "pointsBalance",
    "label": "POINTS BALANCE",
    "value": pointBalance,
    "textAlignment": "PKTextAlignmentNatural", 
    "labelColor": "rgb(57, 133, 57)" 
});
```

```
ValidationError happened. Cannot add field. "labelColor" is not allowed.
```

But autually in apple development guide, it allows the developer to modify the label color.

"**Pass Colors Can Be Customized**
By default, Wallet chooses the colors for your pass background and text. To override a color, provide a value for the appropriate key at the top level of the pass.json file. Colors are specified as RGB values—for example, rgb(0, 255, 0) is bright green. You can customize three colors:
....
The label color, used for the labels of fields shown on the front of the pass. The key for the label color is labelColor."

Thanks in advance!

**Comment by alexandercerutti** [comment, marked as answer]
Hi @SamYueHW, thanks for using passkit-generator!

All labels in the pass must be of the same color. In fact, `labelColor` is a "global" (or "root" or "top level", as you reported) property of the `pass.json`. Hence, **_you cannot customize the single labels_**.

```json
{
    "labelColor": "rgb(...)",
    "headerFields": [ ... ]
}
```

Hope this is helpful!
Let me know if I can help you any further.

**Comment by SamYueHW** [comment]
Thank you so much for guiding me! 
If I would like to reset the value color for header field. Just like someone ‘s pass example below, How I can do that?
![88ae184e96733a762d1242a20939cc9](https://github.com/alexandercerutti/passkit-generator/assets/129275455/3d5fd965-6fda-47a8-8f0d-fc1740d85f8b)

**Reply by alexandercerutti** [reply]
As I said, you cannot reset the color of a single field, without affecting all the others. This statement is valid for both properties `labelColor` and `foregroundColor`.

In your screenshot, `labelColor` represents "Hello" string color, while `foregroundColor` represents "Yue".

You can experiment on https://pkvd.app (Passkit Visual Designer) to see it better.

**Reply by SamYueHW** [reply]
Thank you so much for your quick reply and patient explaination! I got it! Thanks again!

**Reply by alexandercerutti** [reply]
Let me know if you need further assistance. If you could leave a ⭐, it would be awesome!
